### PR TITLE
[8.0] Manage lot directly on stock logistics workflow details

### PR DESCRIPTION
**Pull request by OpenCode**

### CODE REVIEW

File: stock_picking_package_preparation_line/views/stock_picking_package_preparation_line.xml [hunk 1]
```diff
@@ -16,6 +16,7 @@
                             <field name="product_id"/>
                             <field name="product_uom_qty" />
                             <field name="product_uom" groups="product.group_uom"/>
+                            <field name="lot_id"/>
```
Comment (andrea4ever) [inline stock_picking_package_preparation_line/views/stock_picking_package_preparation_line.xml]: Please add "groups=stock.group_production_lot"

File: stock_picking_package_preparation_line/views/stock_picking_package_preparation_line.xml [hunk 2]
```diff
@@ -25,6 +26,10 @@
                                        attrs="{'required': [('product_id', '!=', False)]}"/>
                                 <field name="product_uom" groups="product.group_uom"
                                        attrs="{'required': [('product_id', '!=', False)], 'readonly': [('move_id', '!=', False)]}"/>
+                                <field name="lot_id"
+                                       domain="[('product_id', '=', product_id)]"
+                                       context="{'default_product_id': product_id}"
+                                       attrs="{'invisible': [('product_id', '=', False)], 'readonly': [('move_id', '!=', False)]}"/>
```
Comment (andrea4ever) [inline stock_picking_package_preparation_line/views/stock_picking_package_preparation_line.xml:33]: Please add "groups=stock.group_production_lot"

**Comment by andrea4ever** [comment]
@OpenCode 2 remarks on your commit, for better group management.
Thank you

**Comment by OpenCode** [comment]
@andrea4ever Done! Thank you!

**Comment by andrea4ever** [comment]
:+1:  for me.

**Comment by eLBati** [comment]
Thanks @OpenCode , could you add a test covering the new use case?

**Comment by OpenCode** [comment]
@eLBati Test done

File: stock_picking_package_preparation_line/models/stock_picking_package_preparation_line.py [hunk 3]
```diff
@@ -220,6 +227,19 @@ def action_put_in_pack(self):
...
+                    if line.lot_id:
+                        pack_model.create({
+                            'product_id': line.product_id.id,
+                            'product_uom_id': line.product_uom.id,
+                            'product_qty': line.product_uom_qty,
+                            'lot_id': line.lot_id.id,
+                            'location_id': move_data['location_id'],
+                            'location_dest_id': move_data['location_dest_id'],
+                            'date': datetime.now(),
+                            # 'owner_id': prod.owner_id.id,
```
Comment (eLBati) [inline stock_picking_package_preparation_line/models/stock_picking_package_preparation_line.py]: If not needed, please remove this comment

**Comment by eLBati** [comment]
Just tested on http://3145085-218-c982dc.runbot2.odoo-community.org/ (base DB) but I can't find the lot field in preparation lines.
@OpenCode do you know if it is a runbot or module problem?

**Comment by OpenCode** [comment]
@eLBati there is a problem in runbot. If I open the record of the form view "sppp.line.stock.picking.package.preparation.form" I see a different arch from this PR.

**Comment by eLBati** [comment]
ok, LGTM :+1:

**Comment by gurneyalex** [comment]
:+1:

**Pull request merged**